src/components/Glp/SwapErrorModal.tsx

```diff
@@ -0,0 +1,75 @@
+import { BigNumber } from "@ethersproject/bignumber";
+import { useEffect, useState } from "react";
+import { ethers } from "ethers";
+import { getNativeToken } from "../../config/Tokens";
+import { InfoToken, Token } from "../../domain/tokens/types";
+import Modal from "../Modal/Modal";
+import { get1InchSwapUrl } from "../../domain/common";
+import { getLowestFeeTokenForBuyGlp } from "../../domain/prices";
+
+const { AddressZero } = ethers.constants;
+
+type Props = {
+  swapToken: Token;
+  isVisible: boolean;
+  setIsVisible: () => void;
+  chainId: number;
+  glpAmount: BigNumber;
+  usdgSupply: BigNumber;
+  totalTokenWeights: BigNumber;
+  glpPrice: BigNumber;
+  infoTokens: {
+    [key: string]: InfoToken;
+  };
+};
+
+interface LowestFeeToken extends Token {
+  fees: number;
+}
+
+export default function SwapErrorModal({
+  swapToken,
+  isVisible,
+  setIsVisible,
+  chainId,
+  glpAmount,
+  usdgSupply,
+  totalTokenWeights,
+  glpPrice,
+  infoTokens,
+}: Props) {
+  const [lowestFeeToken, setLowestFeeToken] = useState<LowestFeeToken>();
+  useEffect(() => {
+    const lowestFeeToken = getLowestFeeTokenForBuyGlp(
+      chainId,
+      glpAmount,
+      glpPrice,
+      usdgSupply,
+      totalTokenWeights,
+      infoTokens,
+      swapToken.address
+    );
+    setLowestFeeToken(lowestFeeToken);
+  }, [chainId, glpAmount, glpPrice, usdgSupply, totalTokenWeights, infoTokens, swapToken.address]);
+
+  if (!lowestFeeToken || !swapToken) return "";
```

Comment:
not sure if it makes any different, i usually do `return null;` to not render component. maybe we can use consistent `return null` everywhere in code?

Reply:
Sure!